### DISCUSSION TITLE
Positional and keyword arguments in PyEt? (Examples: penman and pm)

### DISCUSSION THREAD

**Discussion by SorenJessen** [pyet-org/pyet#77] [Q&A] [answered]
I have these data, daily values:
mean wind speed ('wind' in PyEt terminology, m/s)
mean temperature ('tmean', deg. C)
min temperature ('tmin', deg. C)
max temperature ('tmax', deg. C)
mean relative humidity ('rh', %)
sum global radiation ('rs', MJ/m2)
sum sunshine hours ('n', hours)

I also have the site latitude and elevation.

Now how do I best make use of my data in pyet to calculate PET using the penman-monteith (pm) and penman (penman) methods?

For example, if I call the penman method:
pet_penman = pyet.penman(which positional arguments? which keyword arguments?)

Thanks!

**Comment by mvremec** [comment, marked as answer]
Hi @SorenJessen ,

Thank you for sharing the details of your dataset. You've got all the necessary data to effectively use both the penman and penman-monteith methods in PyEt. Here's a breakdown of how you can proceed:

For both the Penman and Penman-Monteith methods in PyEt, the tmean (mean temperature) and wind (wind speed) are positional arguments. All other variables should be passed as keyword arguments. This flexibility allows PyEt to adapt to various datasets like yours.

Example Usage:
Here’s how you can use your data with both methods:
`pet_penman = pyet.penman(tmean, wind, rs=rs, tmax=tmax, tmin=tmin, rh=rh, elevation=elevation, latitude=latitude)`
`pet_pm = pyet.pm(tmean, wind, rs=rs, tmax=tmax, tmin=tmin, rh=rh, elevation=elevation, latitude=latitude)`

*Regarding the sunshine hours data:*
Since you have the sum of global radiation (rs), you can omit the sunshine hours for the Penman equation calculations. The Net radiation component can be input directly using rs, or calculated internally if rs isn't available but sunshine hours are provided. For your dataset, using rs directly is recommended.

*Additional Resources:*
I recommend checking out the example [notebooks](https://pyet.readthedocs.io/en/dev/examples/06_worked_examples_McMahon_etal_2013.html) and the function docstrings or in our [ReadTheDocs](https://pyet.readthedocs.io/en/latest/api/generated/generated/pyet.combination.penman.html#pyet.combination.penman) for more details on the usage of these methods and the roles of different parameters. I’ll also be adding a new Introductory Notebook soon to demonstrate different keyword arguments you can use for each function.

Thank you for your question, and don't hesitate to reach out if you need further assistance!

Best,
Matevz

**Reply by SorenJessen** [reply]
Thanks so much for prompt reply!
